File: chapter3/py/src/example1.py
# In Python

# Import method
from pyspark.context import SparkContext

# Create SparkContext object
sc = SparkContext('local', 'test')

# Create an RDD of tuples (name, age)
dataRDD = sc.parallelize([("Brooke", 20), ("Denny", 31), ("Jules", 30), ("TD", 35), ("Brooke", 25)])
# Use map and reduceByKey transformations with their lambda
# expressions to aggregate and then compute average
print("Prueba")
print(dataRDD.glom().collect())
print("Prueba")
agesRDD = (dataRDD
	.map(lambda x: (x[0], (x[1], 1)))
	.reduceByKey(lambda x, y: (x[0] + y[0], x[1] + y[1]))
	.map(lambda x: (x[0], x[1][0]/x[1][1])))
print(agesRDD.glom().collect())

File: chapter3/scala/script-examples/example1.scala
// In Scala
import org.apache.spark.sql.functions.avg
import org.apache.spark.sql.SparkSession

// Create a DataFrame using SparkSession
val spark = SparkSession.builder.appName("AuthorsAges").getOrCreate()

// Create a DataFrame of names and ages
val dataDF = spark.sparkContext.parallelize(Seq(("Brooke", 20), ("Brooke", 25),
	("Denny", 31), ("Jules", 30), ("TD", 35))).toDF("name", "age")

// Group the same names together, aggregate their ages, and compute an average
val avgDF = dataDF.groupBy("name").agg(avg("age"))

// Show the results of the final execution
avgDF.show()